viewer/core/src/public/migration/Cognite3DViewer.ts

```diff
@@ -457,6 +595,54 @@ export class Cognite3DViewer {
     }
   }
 
+  /**
+   * Sets camera controls mode anything allowed in CameraControlsOptions type.
+   * @param controlsOptions JSON object with camera controls options.
+   */
+  setCameraControlsMode(controlsOptions: CameraControlsOptions) {
+    this._cameraControlsOptions.useOnClickTargetChange =
+      controlsOptions?.onClickTargetChange ?? this._cameraControlsOptions.useOnClickTargetChange;
+    this._cameraControlsOptions.canInterruptAnimations =
+      controlsOptions?.canInterruptAnimations ?? this._cameraControlsOptions.canInterruptAnimations;
+
+    switch (controlsOptions?.zoomToCursor) {
+      case 'disable':
+        this._cameraControlsOptions.useScrollTargetControls = false;
+        this.controls.zoomToCursor = false;
+
+        if (this._cameraControlsOptions.scrollEventsAdded) {
+          this.setupOtherControlsModes(true);
+        }
+        break;
+      case 'basicLerp':
+        this._cameraControlsOptions.useScrollTargetControls = false;
+        this.controls.useScrollTarget = false;
+
+        if (this._cameraControlsOptions.scrollEventsAdded) {
+          this.setupOtherControlsModes(true);
+        }
+        break;
+      case 'scrollTarget':
+        this._cameraControlsOptions.useScrollTargetControls = true;
+        this.controls.useScrollTarget = true;
+
+        if (!this._cameraControlsOptions.scrollEventsAdded) {
+          this.setupOtherControlsModes();
+        }
+        break;
+      default:
+        break;
```

Comment:
use `assertNever(controlsOptions.zoomToCursor)` to ensure you handle all valid enum values.